spec/rubocop/cop/rails/active_support_on_load_spec.rb

```diff
@@ -0,0 +1,114 @@
+# frozen_string_literal: true
+
+RSpec.describe(RuboCop::Cop::Rails::ActiveSupportOnLoad, :config) do
+  it 'adds offense when trying to extend a framework class with include' do
+    expect_offense(<<~RUBY)
+      ActiveRecord::Base.include(MyClass)
+      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Use `ActiveSupport.on_load(:active_record) { include MyClass }` instead of `ActiveRecord::Base.include(MyClass)`.
+    RUBY
```

Comment:
Can you use `expect_correction` for autocorrection testing?

```suggestion
    RUBY

    expect_correction(<<~RUBY)
      ActiveSupport.on_load(:active_record) { include MyClass }
    RUBY
```

Other `expect_offense`s are similar.